### PR TITLE
fix(docs): documented CJS tree shaking solution for icons and react-core

### PR DESCRIPTION
**Pull request by evwilkin**
**What**: Addresses #3268 

**Additional issues**:

### CODE REVIEW

File: packages/patternfly-4/react-core/README.md [hunk 1]
```diff
@@ -112,3 +112,52 @@ yarn test packages/patternfly-4/react-core
...
+
+Use ESM module imports to enable tree shaking with no additional setup required.
+```JS
+import { TimesIcon } from '@patternfly/react-icons';
+```
+
+To enable tree shaking with named imports for CJS modules, utilize [babel-plugin-transform-imports](https://www.npmjs.com/package/babel-plugin-transform-imports) and update a babel.config.js file to utilize the plugin:
+```JS
+require.extensions['.css'] = () => undefined;
+const components = require('@patternfly/react-core/dist/js/components');
```
Comment (jschuler) [inline packages/patternfly-4/react-core/README.md]: our package.json files have `main` and `module` entries, when doing require it should find the cjs entry point as specified in `main` and when doing imports it should find the esm entry point in `module`. So I don't think you need the whole path?

Comment (jschuler) [reply packages/patternfly-4/react-core/README.md]: anyways, might be different for babel js resolution. looks good :)

**Comment by Hyperkid123** [comment]
@evwilkin @dlabaj @redallen Hello, I did some testing and there are some issues with this particular config. Adding link to a issue with more detail: https://github.com/patternfly/patternfly-react/issues/3268#issuecomment-566442100

**Comment by martinpovolny** [comment]
@evwilkin : First on all I'd like to thank you for keeping working on this. I really appreciate that despite what I am going to write next.

So I have to say that this issue makes be wonder about Patternfly development processes.

You have an issue, that is complex, hard to fix, hard to understand, hard to discuss.

You discuss it for weeks. Have ideas, some wrong, some right, obviously it's not a simple thing.

The issue was already once prematurely closed and reopened.

Then you make a PR and merge it within 1 hour. Merge it within one hour since it was opened w/o giving the people who where previously discussing the issue a chance to check it/test it/review it.

Well and no. It did not resolve the issue.

Wow. Just wow.

**Comment by evwilkin** [comment]
@martinpovolny please note that we've kept the original issue open (#3268 ) until we're able to validate together that we have a successful fix, and are happy to continue working together to resolve this.

**Comment by Hyperkid123** [comment]
@evwilkin I tried to apply the transform plugin to the react-core repo to use full imports on react icons and the results are great:

```

Compiled successfully.

Automatically optimizing pages  

Page                    Size
┌ ○ /                   27 kB
└   /_app               1.11 kB
+ shared by all         71.8 kB
  ├ chunks/commons.js   65.7 kB
  ├ runtime/main.js     5.26 kB
  └ runtime/webpack.js  812 B

λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)
●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)

Done in 4.31s.
```

The bundle now contains only the components that it should. I will create PR with the changes to the react-core repo.

**Pull request merged**